Build SWAT+ with CMake
Based on the work by @changliao1025 in https://github.com/changliao1025/swat_hpc
author: Alex Domingo (Vrije Universiteit Brussel)
--- CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ CMakeLists.txt	2020-11-05 01:03:54.740033000 +0100
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.16)
+
+project(swatplus Fortran)
+
+if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE Release)
+endif()
+
+add_subdirectory(source_codes)
+
+install(DIRECTORY database_files example_data_descriptions modular_database_table data documentation revision_notes
+        DESTINATION share)
--- source_codes/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ source_codes/CMakeLists.txt	2020-11-05 00:57:43.224307000 +0100
@@ -0,0 +1,8 @@
+file(GLOB SWAT_SRCS *.f90 *.F90)
+
+add_executable(swatplus ${SWAT_SRCS})
+
+SET(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -ffree-line-length-none -ffixed-line-length-none")
+SET(CMAKE_Fortran_FLAGS_RELEASE "")
+
+install(TARGETS swatplus DESTINATION bin)
